### PR TITLE
docs: Change structure of document

### LINKED ISSUE #62: Enhancing Document Readability
**Current Documentation Challenges**

The current documentation for the overlay-kit is scattered across multiple sources. Users must read through various documents to fully understand how to use the overlay-kit, and there are significant risks of missing important information, such as precautions and best practices.

**Proposed Solution**

This issue aims to consolidate and enhance the documentation to make it more intuitive and accessible. The goal is to ensure that users can understand what the overlay-kit is designed to achieve without having to look at example code. We need to present a clear, unified view of the tool’s functionality and usage guidelines in one comprehensive document.

### PR DESCRIPTION
**Pull request by jungpaeng**
## Description

**Related Issue:** Fixes #62

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation update

## Checklist

- [ ] I have performed a self-review of my own code.
- [ ] My code is commented, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

File: docs/ko/usage/open-overlay.md [hunk 1]
```diff
@@ -29,7 +25,9 @@ overlay.open(({ isOpen, close, unmount }) => {
 오버레이를 닫기 위해서는, 콜백 함수에 주어진 `close` 함수를 사용하세요. `close` 함수를 호출하면, `isOpen` 상태는 `false`가 되어요. 오버레이가 `open` Prop을 구현한다면, 화면에서 사라지죠.
 
 대부분의 오버레이는 닫힐 때 애니메이션을 가지고 있어요. overlay-kit에서는 오버레이를 닫아도 애니메이션이 계속 실행되도록, Mount 상태를 유지해요.
-메모리 릭을 피하기 위해서는, 닫히는 애니메이션이 끝난 다음에 잊지 않고 오버레이를 Unmount시켜줘야 해요.
+
+> [!WARNING]
+> 메모리 릭을 피하기 위해서는, 닫히는 애니메이션이 끝난 다음에 잊지 않고 오버레이를 Unmount시켜줘야 해요.
```
Comment (ojj1123) [inline docs/ko/usage/open-overlay.md:30]: NIT: How about linking relevant issue?

https://github.com/toss/overlay-kit/issues/55

File: docs/ko/quick-start/mui.md [hunk 2]
```diff
@@ -144,27 +142,43 @@ function Example() {
...
-});
-```
+위의 예제를 `overlay.openAsync(...)`로 다시 구성해볼게요.
+
+```tsx{5-24}
+function Example() {
+  return (
+    <button
+      onClick={async () => {
+        const agreed = await overlay.open<boolean>(({ isOpen, close }) => {
```
Comment (jgjgill) [inline docs/ko/quick-start/mui.md:158]: I think `openAsync`, not `open`

```suggestion
        const agreed = await overlay.openAsync<boolean>(({ isOpen, close }) => {
```

File: docs/ko/usage/open-overlay.md (hunk 1 above)
Comment (manudeli) [inline docs/ko/usage/open-overlay.md:30]:
```suggestion
> 메모리 누수을 피하기 위해서는, 닫히는 애니메이션이 끝난 다음에 잊지 않고 오버레이를 Unmount시켜줘야 해요.
```

```suggestion
> Memory leak을 피하기 위해서는, 닫히는 애니메이션이 끝난 다음에 잊지 않고 오버레이를 Unmount시켜줘야 해요.
```

**Comment by jungpaeng** [comment]
Our team is currently in the process of reviewing the document. Once the review is complete, significant changes to the structure are likely.

I will incorporate your comments after this task is finished. Thank you.

**Comment by jungpaeng** [comment]
Continue to #67

**Pull request closed without merging**